hbase-common/src/main/java/org/apache/hadoop/hbase/ScheduledChore.java

```diff
@@ -183,18 +176,21 @@ public void run() {
       if (LOG.isInfoEnabled()) LOG.info("Chore: " + getName() + " was stopped");
     } else {
       try {
+        // TODO: Histogram metrics per chore name.
+        // For now, just measure and log if DEBUG level logging is enabled.
+        long start = 0;
+        if (LOG.isDebugEnabled()) {
+          start = System.nanoTime();
```

Comment:
A question of clarification. I think what you have is correct, but I'm asking anyway. We have `EnvironmentEdgeManager`, which abstracts a form of clock. `System.nanoTime()` is a different form of clock. Is there some case where we'd want the `nanoTime` to be abstracted behind `EnvironmentEdgeManager`? My understanding is no, we don't have such a use-case, because `EnvironmentEdgeManager` is for abstracting over a clock-time (a la `java.time.Instant`), while `nanoTime` is providing a timer for calculating a time duration, something that is abstract from a clock-time.

Reply:
We don't, but we could. Would be another scope of change I think.

Reply:
Yeah I was kind of caviler about the ifdebug guards. Will fix

Reply:
I'm not asking for a scope change, just clarifying my understanding.